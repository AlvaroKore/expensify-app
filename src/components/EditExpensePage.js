import React from 'react';
import { connect } from 'react-redux';
import { editExpense, removeExpense } from '../actions/expenses';
import ExpenseForm from './ExpenseForm';

export class EditExpensePage extends React.Component {
	onSubmit = (expense) => {
		this.props.editExpense(this.props.expense.id, expense);
		this.props.history.push('/');
	};

	removeExpense = (e) => {
		e.preventDefault();
		this.props.removeExpense(this.props.expense.id);
		this.props.history.push('/');
	};

	render() {
		return (
			<div>
				<ExpenseForm expense={this.props.expense} onSubmit={this.onSubmit} />
				<button onClick={this.removeExpense}>Remove</button>
			</div>
		);
	}
}

const mapStateToProps = (state, props) => {
	return {
		expense: state.expenses.find((expense) => props.match.params.id === expense.id)
	};
};

const mapDispatchToProps = (dispatch) => ({
	editExpense: (id, expense) => dispatch(editExpense(id, expense)),
	removeExpense: (id) => dispatch(removeExpense({ id }))
});
export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
