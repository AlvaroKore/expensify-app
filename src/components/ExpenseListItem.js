import React from 'react';
import { Link } from 'react-router-dom';

const ExpenseListItem = (props) => (
	<div>
		<h3>
			<Link to={`/edit/${props.id}`}>{props.description}</Link>
		</h3>
		<p>amount: {props.amount}</p>
		<p>note: {props.note}</p>
		<p>createdAt: {props.createdAt}</p>
		<hr />
	</div>
);

export default ExpenseListItem;
