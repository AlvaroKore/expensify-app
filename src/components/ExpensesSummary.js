import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import getExpensesTotal from '../selectors/expenses-total';


export class ExpensesSummary extends React.Component {
    render () {
        return (
            <div>
                <h3>ExpensesSummary</h3>
                <p>
                    Viewing {this.props.expenseCount} expenses totalling {this.props.expensesTotal}
                </p>
            </div>
        )
    }
}


const mapStateToProps = (state) => {
	const expenses = selectExpenses(state.expenses, state.filters);
	return {
		expenseCount: expenses.length,
		expensesTotal: getExpensesTotal(expenses)
	};
};

export default connect(mapStateToProps)(ExpensesSummary);
