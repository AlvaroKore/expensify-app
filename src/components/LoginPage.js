import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

const LoginPage = ({ startLogin }) => (
	<div>
		<h1>Login Page</h1>

		<button onClick={startLogin} type="submit">
			Login
		</button>
	</div>
);

const mapDispatchToProps = (dispatch) => {
	return {
		startLogin:  () => dispatch(startLogin())
	};
};

export default connect(undefined, mapDispatchToProps)(LoginPage);
