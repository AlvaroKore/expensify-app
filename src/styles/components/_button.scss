.button {
	background: $blue;
    border: none;
    display: inline-block;
    text-decoration: none;
    line-height: 1;
    color: white;
    font-size: $font-size-large;
    font-weight: 300;
    padding: $s-size;

}

.button--link {
    background: none;
    
}

.button--secondary {
    background: #888888;
}
