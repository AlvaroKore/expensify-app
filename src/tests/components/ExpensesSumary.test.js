import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render ExpensesSummary correctly', () => {
	const wrapper = shallow(<ExpensesSummary expenseCount = {1} expensetotal={288} />);
	expect(wrapper).toMatchSnapshot();
});
