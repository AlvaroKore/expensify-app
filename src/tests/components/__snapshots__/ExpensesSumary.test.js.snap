// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary correctly 1`] = `
<div>
  <h3>
    ExpensesSummary
  </h3>
  <p>
    Viewing 
     expenses totalling 
  </p>
</div>
`;
